liquidctl/driver/aura_led.py

```diff
@@ -0,0 +1,322 @@
+        status = []
+        # Get config table
+        self._write(_FUNCTION_CODE["config"])
+        data = self.device.read(_READ_LENGTH)
+        if data[1] == 0x30:
+            start_index = 4  # index of first record
+            num = 6  # number of bytes per record
+            count = 1
+            while start_index + num < _READ_LENGTH:
+                status.append(
+                    (
+                        "Device Config: " + str(count),
+                        ", ".join(
+                            "0x{:02x}".format(x) for x in data[start_index : start_index + num]
+                        ),
+                        "",
+                    )
+                )
+                start_index += num
+                count += 1
+        else:
+            status.append("Unexpected reply for config", "", "")
+        return status
+
+    def set_color(self, channel, mode, colors, speed="normal", **kwargs):
+        """Set the color mode for a specific channel.
+
+        `colors` should be an iterable of zero or one `[red, green, blue]`
+        triples, where each red/green/blue component is a value in the range
+        0–255.
+        """
+        colors = iter(colors)
+        if _COLOR_MODES[mode].takes_color:
+            try:
+                r, g, b = next(colors)
+                single_color = (r, g, b)
+            except:
+                raise ValueError(f"one color required for this mode") from None
+        else:
+            single_color = (0, 0, 0)
+
+        if channel != "sync" and channel not in _COLOR_CHANNELS:
+            # _LOGGER.error("channel %s not valid", channel)
+            message = "valid channels are "
+            for chan in _COLOR_CHANNELS:
+                message += chan + " "
+            # _LOGGER.error(message)
+            raise KeyError(message) from None
+            return
+        """
+        This is experimental (it's an example of direct mode)
+        if mode == 'off':
+            self.channel_off(channel)
+            self.reset_all_channels()
+            return
+        """
+        if channel == "sync":
+            selected_channels = _COLOR_CHANNELS.values()
+        else:
+            selected_channels = (_COLOR_CHANNELS[channel],)
+        full_cmd_seq = []  # entire series of commands are added to this list
+        """
+        Experimental code for treating RGB channel differently from others
+        if channel == "rgb":
+            cmd_tuple=self.construct_color_commands(channel, mode, single_color)
+            self._write(cmd_tuple[0])
+            self._write(cmd_tuple[1])
+            self._write(_FUNCTION_CODE["end_seq2"])
+            self.end_color_sequence()
+            self._write(cmd_tuple[0])
+            self._write(cmd_tuple[1])
+            self._write(_FUNCTION_CODE["end_seq2"])
+        else:
+        """
+        for chan in selected_channels:
+            cmd_tuple = self.construct_color_commands(chan.name, mode, single_color)
```

Comment:
What happens if you set different modes / colors here for each of the channels? Would that allow each channel to have an independent effect?

Reply:
Once or twice I found that it was possible to set different color modes to different channels -- even in "effect" mode. I'll need to run more tests to see if I can reproduce the behavior. It may be possible to set RGB channel independently of ARGB channels, but again, need to re-test.